src/editor/hole.ts

```diff
@@ -49,47 +50,23 @@ export class Hole {
             )}
 
             code.subscribe(
-                CallbackType.loseFocus,
-                new Callback(() => {
-                    this.element.classList.remove(Hole.editableHoleClass);
-
-                    if (code instanceof IdentifierTkn) {
-                        Hole.holes.forEach((hole) => {
-                            if (hole.element.classList.contains(Hole.inScopeHole)) {
-                                hole.element.classList.remove(Hole.inScopeHole);
-                            }
-                        });
-                    }
-                })
-            );
-
-            code.subscribe(
-                CallbackType.focus,
+                CallbackType.focusEditableHole,
                 new Callback(() => {
                     this.element.classList.add(Hole.editableHoleClass);
-
-                    if (code instanceof IdentifierTkn) {
-                        Hole.holes.forEach((hole) => {
-                            if (
-                                hole.code instanceof TypedEmptyExpr &&
-                                ((code.getParentStatement() as VarAssignmentStmt).dataType ==
-                                    (hole.code as TypedEmptyExpr).type ||
-                                    (hole.code as TypedEmptyExpr).type == DataType.Any) &&
-                                hole.code.getParentStatement().hasScope() &&
-                                hole.code
-                                    .getParentStatement()
-                                    .scope.isValidReference(
-                                        (code.getParentStatement() as VarAssignmentStmt).buttonId,
-                                        hole.code.getSelection().startLineNumber
-                                    )
-                            ) {
-                                hole.element.classList.add(Hole.inScopeHole);
-                            }
-                        });
-                    }
                 })
             );
         }
+        else if(code instanceof TypedEmptyExpr){
```

Comment:
This code highlights identifier holes of valid variable insertions to this TypedEmptyExpr